device-registry/src/main/scala/org/genivi/sota/device_registry/DeviceRepository.scala

```diff
@@ -196,4 +225,28 @@ object Devices {
     _ <- systemInfos.insertOrUpdate(SystemInfo(id,data))
   } yield ()
 
+  def findGroupInfoByName(groupName: String, ns: Namespace)(implicit ec: ExecutionContext): DBIO[GroupInfoType] = {
+    groupInfos
+      .filter(r => r.namespace === ns && r.groupName === groupName)
+      .result
+      .headOption
+      .flatMap(_.fold[DBIO[GroupInfoType]](DBIO.failed(Errors.MissingGroupInfo))(x => DBIO.successful(x.groupInfo)))
+  }
+
+  def createGroupInfo(groupName: String, namespace: Namespace, data: GroupInfoType)(implicit ec: ExecutionContext)
+      : DBIO[Unit] =
+    for {
+      _ <- existsGroup(groupName, namespace).asTry.flatMap {
```

Comment:
This is what I mean https://github.com/advancedtelematic/rvi_sota_server/blob/feature/optional_device_name/common/src/main/scala/org/genivi/sota/db/SlickExtensions.scala#L58

You can then use it like this:

https://github.com/advancedtelematic/rvi_sota_server/blob/feature/optional_device_name/device-registry/src/main/scala/org/genivi/sota/device_registry/DeviceRepository.scala#L62

Provided you setup unique keys properly.

You can just copy that code if you want, I'll handle conflict errors on my branch.